astropy/utils/data.py

```diff
@@ -62,6 +62,26 @@
 _dataurls_to_alias = {}
 
 
+class _NonClosingBufferedReader(io.BufferedReader):
+    def __del__(self):
```

Comment:
This is too magical for me. Perhaps @embray can have a look as well here.

Reply:
The more eyes the better.  This was the solution I got from `click` (and it works).  I had to apply it to both `TextIOWrapper` and `BufferedRead`.  Some references:  
* https://bugs.python.org/issue21363
* https://github.com/pallets/click/blob/9da166957f5848b641231d485467f6140bca2bc0/src/click/_compat.py#L61